objects/src/transaction/tx_progress.rs

```diff
@@ -43,78 +44,112 @@ impl TransactionProgress {
     // STATE MUTATORS
     // --------------------------------------------------------------------------------------------
 
-    pub fn start_prologue(&mut self, cycle: u32) {
+    pub fn start_prologue(&mut self, cycle: usize) {
```

Comment:
Not sure we should use `usize` here (and in other similar places). If we do want to change it, I wonder if we should change it to `RowIndex` type from `miden-vm`.